lib/reply.js

```diff
@@ -140,27 +143,36 @@ function wrapOnSendEnd (err, reply, payload) {
 }
 
 function onSendEnd (reply, payload) {
+  const res = reply.res
+
   if (payload === undefined || payload === null) {
     reply.sent = true
-    reply.res.end()
+    reply._headers['content-length'] = '0'
```

Comment:
The `Content-Length` header should not be set here. There's rules about it in [RFC 7230](https://tools.ietf.org/html/rfc7230#section-3.3.2).